engine/src/main/java/org/hibernate/search/query/engine/impl/EntityInfoImpl.java

```diff
@@ -62,31 +66,21 @@ public String getIdName() {
 	}
 
 	@Override
-	public List<Integer> getIndexesOfThis() {
-		return indexesOfThis;
-	}
-
-	@Override
-	public boolean isProjectThis() {
-		return indexesOfThis.size() != 0;
+	public Object getEntityInstance() {
+		for ( Object obj : projection ) {
+			if ( clazz.isInstance( obj ) ) {
+				return obj;
```

Comment:
I think I'd rather store the instance as a field than to have to loop through the projections with instanceof attempts?

You can easily store the EntityInstance during `populateWithEntityInstance`.